src/i18n/TextCollection/i18nTextCollector.php

```diff
@@ -210,9 +212,9 @@ public function collect($restrictToModules = [], $mergeWithExisting = false)
         // Restrict modules we update to just the specified ones (if any passed)
         if (!empty($restrictToModules)) {
             // Normalise module names
-            $modules = array_filter(array_map(function ($name) {
-                $module = ModuleLoader::inst()->getManifest()->getModule($name);
-                return $module ? $module->getName() : null;
+            $allModules = $this->getModulesAndThemes();
+            $modules = array_filter(array_map(function ($name) use ($allModules) {
+                return array_key_exists($name, $allModules) ? $name : null;
```

Comment:
Here we're relying directly on the array keys in returned from `getModulesAndThemes()`, but in `getEntitiesByModule()` we only rely on array keys for themes - we should be consistent and either always use the array key, or only ever use the array key for themes.

Reply:
I wanted to keep as much of the existing functionality intact and only rely on the keys where needed. I can't see the woods for the trees. How would you change this here?

Reply:
Hey @GuySartorelli what do you want me to do here to get this over the line? Cheers

Reply:
```suggestion
                return array_key_exists($name, $allModules) ? $this->getModuleName($name, $module) : null;
```

```php
private function getModuleName(string $origName, Module $module)
{
    return strpos($origName, self::THEME_PREFIX) === 0 ? $origName : $module->getName();
}
```

Reply:
where do I get the `$module` parameter from? Sprry, too many nested methods for bmy brain...

Reply:
The `getModulesAndThemes()` method returns an array of `$name => $module` doesn't it?

Reply:
Yes, correct. Thanks!